sentry-core/src/session.rs

```diff
@@ -40,10 +41,23 @@ pub struct Session {
     dirty: bool,
 }
 
+impl Drop for Session {
+    fn drop(&mut self) {
+        self.close();
+        if let Some(item) = self.into_envelope_item() {
+            let mut envelope = Envelope::new();
+            envelope.add(item);
+            self.client.capture_envelope(envelope);
+        }
+    }
+}
+
 impl Session {
     pub fn from_stack(stack: &StackLayer) -> Option<Self> {
-        let options = stack.client.as_ref()?.options();
+        let client = stack.client.as_ref()?;
+        let options = client.options();
         Some(Self {
+            client: client.clone(),
```

Comment:
```suggestion
        let client = stack.client.clone()?;
        let options = client.options();
        Some(Self {
            client,
```

Reply:
nope ;-) you can’t access anything from `options` *after* you have moved `client` into the new struct :-D